### PR TITLE
NormalizeL2 - reference implementation revision

### PR DESCRIPTION
**Pull request by mitruska**
### Details: 
NormalizeL2 - reference implementation revision

Reference implementation changes:
- [x] Align reference implementation for empty `axes` case 
(fix CPU [skipped SLT "incorrect reference"](https://github.com/openvinotoolkit/openvino/pull/6349/files#diff-67343461a309cd31117af56e5fb233c7cf8caaa36ba134bced01806835b57a57L55-L56), corresponding to [mkldnn_normalize_node.cpp#L1311](https://github.com/openvinotoolkit/openvino/pull/6349/files#diff-67343461a309cd31117af56e5fb233c7cf8caaa36ba134bced01806835b57a57L55-L56), 
but it needs skipping current empty axes SLT case for GPU @vladimir-paramuzov)

~~MKLDNN change:~~  (Reverted, ticket 59788)
~~Apply eps before `sqrt` (commit https://github.com/openvinotoolkit/openvino/pull/6349/commits/ea62dd9e808a3afd8f8d0680ac0792fa4535b0e5)
(aligned with spec and tf/mxnet, it repairs output mismatch for bigger eps or smaller tolerance)~~

Tests changes:
- [x] Update single layer tests
- [x] Update backend tests
- [x] Add to verified ops list

### Tickets:
 - 40725 (45807, 45808)

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@chenhu-wang Could you please review changes in CPU plug-in code. Looks like we have different order of execution comparing with the spec.

**Comment by chenhu-wang** [comment]
> @chenhu-wang Could you please review changes in CPU plug-in code. Looks like we have different order of execution comparing with the spec.

1. In general, eps is aimed to avoid divide zero, from safety perspective it should be applied exactly before divide(after sqrt). Some computation after applying eps will lead to zero value again before divide. This is the behavior of pytorch norm:https://pytorch.org/docs/master/generated/torch.nn.functional.normalize.html#torch.nn.functional.normalize.
2. Given pytorch and TF/mxnet behavior is not the same. We need add new mode or update current eps_mode attribute to cover all cases. Current eps_mode only cover add or max options, not cover apply before sqrt or after sqrt options. We need evaluate if update current operation spec is safe or we need add new version of this operation.
3. Current [spec](https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/normalization/NormalizeL2_1.md) itself is not aligned. In attributes description, `eps is the number to be added/maximized to/with the variance` and `Specifies how eps is combined with L2`, variance and L2 is value after sqrt, which say apply eps after sqrt. But in detail description, `output[i0, i1, ..., iN] = x[i0, i1, ..., iN] / sqrt(eps_mode(sum[j0,..., jN](x[j0, ..., jN]**2), eps))`, which indicate apply eps before sqrt.

**Comment by mitruska** [comment]
@chenhu-wang Thank you for the review and valuable comment.
I agree and as you mentioned Pytorch behaviour, I'm going to revert the change for mkldnn (commit ea62dd9).

But there is still inconsistency in mkldnn normalize_nchw between
`across_spatial==true` (epsApply after sqrt)

https://github.com/openvinotoolkit/openvino/blob/5cee8bbf29797f4544b343e803de957e9f041f92/inference-engine/src/mkldnn_plugin/nodes/mkldnn_normalize_node.cpp#L976-L977

and `across_spatial==false` (epsApply before sqrt)
https://github.com/openvinotoolkit/openvino/blob/5cee8bbf29797f4544b343e803de957e9f041f92/inference-engine/src/mkldnn_plugin/nodes/mkldnn_normalize_node.cpp#L1016

The spec is going to be clarified with a separate PR (https://github.com/openvinotoolkit/openvino/pull/6512)
We can't add new mode within the same version of the operation, but we should unify the order of `eps` applying.
As you suggest we can consider epsApply after sqrt as a safer option, 
eps values are usually so small that it's hard to spot the difference.

**Comment by chenhu-wang** [comment]
@mitruska, IMO, as we agree to select a unified eps mode(either before or after sqrt) for normalize-1 with revised spec. We can align this manner within OV across all components. If there would be accuracy issues in the future due to our selected default eps mode, then we need add new version for normalize. For mkl-dnn plugin, now it's ok to keep your changes and need align all relevant epsApply: [1](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_normalize_node.cpp#L976), [2](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_normalize_node.cpp#L1016), 3[https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_normalize_node.cpp#L1060], [4](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_normalize_node.cpp#L1091), [5](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_normalize_node.cpp#L1158), [6](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_normalize_node.cpp#L1193), [7](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_normalize_node.cpp#L1256), [8](https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_normalize_node.cpp#L1293).

File: ngraph/core/reference/include/ngraph/runtime/reference/normalize_l2.hpp [hunk 1]
```diff
@@ -25,12 +25,16 @@ namespace ngraph
                 AxisSet axes = reduction_axes;
```
Comment (pelszkow) [inline ngraph/core/reference/include/ngraph/runtime/reference/normalize_l2.hpp:25]: Don't see reason do copy `reduction_axes` now.

Comment (mitruska) [reply ngraph/core/reference/include/ngraph/runtime/reference/normalize_l2.hpp]: Right, removed.

**Comment by mitruska** [comment]
@chenhu-wang I removed mkldnn changes from this PR and I've created tickets to handle mkldnn issues separately.

**Review by pelszkow — APPROVED** [review]

**Review by iefode — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Review by jdanieck — APPROVED** [review]
